sharktank/sharktank/types/theta.py

```diff
@@ -114,16 +114,18 @@ def transform(self, *transforms: InferenceTensorTransform) -> "Theta":
     def to(self, *, device: Optional[Union[str, torch.device]] = None) -> "Theta":
         return self.transform(InferenceTensorTransforms.to_device(device))
 
-    def pop(self, *name_path: str | int) -> "Theta":
+    def pop(self, *name_path: str | int, inplace: bool = True) -> "Theta":
```

Comment:
I think I left a comment on my previous review, but could not find it. I am posting it again.
The `pop` function seems weird that it would not change the contents.
What we are doing is filtering on a subpath.
I think It would be more clear if there as a function `match` or `match_subpath`.
```
def match(self, *name_path: str | int) -> "Theta"
```